pkg/kapis/logging/v1alpha2/handler.go

```diff
@@ -1 +1,190 @@
 package v1alpha2
+
+import (
+	"github.com/emicklei/go-restful"
+	"kubesphere.io/kubesphere/pkg/api"
+	"kubesphere.io/kubesphere/pkg/models/logging"
+	"kubesphere.io/kubesphere/pkg/simple/client/k8s"
+	loggingclient "kubesphere.io/kubesphere/pkg/simple/client/logging"
+	util "kubesphere.io/kubesphere/pkg/utils/stringutils"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	LevelCluster = iota
+	LevelContainer
+
+	// query type, default to `query`
+	TypeStat   = "statistics"
+	TypeHist   = "histogram"
+	TypeExport = "export"
+)
+
+type handler struct {
+	k8s.Client
+	logging.LoggingOperator
+}
+
+func newHandler(k k8s.Client, l loggingclient.Interface) *handler {
+	return &handler{k, logging.NewLoggingOperator(l)}
+}
+
+func (h handler) handleClusterQuery(req *restful.Request, resp *restful.Response) {
+	Get(req, LevelCluster, h.Client, h.LoggingOperator, resp)
+}
+
+func (h handler) handleContainerQuery(req *restful.Request, resp *restful.Response) {
+	Get(req, LevelContainer, h.Client, h.LoggingOperator, resp)
+}
+
+func Get(req *restful.Request, lvl int, client k8s.Client, loggingOperator logging.LoggingOperator, resp *restful.Response) {
+	typ := req.QueryParameter("type")
+
+	noHit, sf, err := NewSearchFilter(client, lvl, req)
+	if err != nil {
+		api.HandleBadRequest(resp, err)
+	}
+	if noHit {
+		handleNoHit(typ, resp)
+		return
+	}
+
+	switch typ {
+	case TypeStat:
+		res, err := loggingOperator.GetCurrentStats(sf)
+		if err != nil {
+			api.HandleInternalError(resp, err)
+		}
+		resp.WriteAsJson(res)
+	case TypeHist:
+		interval := req.QueryParameter("interval")
+		res, err := loggingOperator.CountLogsByInterval(sf, interval)
+		if err != nil {
+			api.HandleInternalError(resp, err)
+		}
+		resp.WriteAsJson(res)
+	case TypeExport:
+		resp.Header().Set(restful.HEADER_ContentType, "text/plain")
+		resp.Header().Set("Content-Disposition", "attachment")
+		err := loggingOperator.ExportLogs(sf, resp.ResponseWriter)
+		if err != nil {
+			api.HandleInternalError(resp, err)
+		}
+	default:
+		from, _ := strconv.ParseInt(req.QueryParameter("from"), 10, 64)
+		size, err := strconv.ParseInt(req.QueryParameter("size"), 10, 64)
+		if err != nil {
+			size = 10
+		}
+		order := req.QueryParameter("sort")
+		if order != loggingclient.Ascending {
+			order = loggingclient.Descending
+		}
+		res, err := loggingOperator.SearchLogs(sf, from, size, order)
+		if err != nil {
+			api.HandleInternalError(resp, err)
+		}
+		resp.WriteAsJson(res)
+	}
+}
+
+func NewSearchFilter(client k8s.Client, level int, req *restful.Request) (bool, loggingclient.SearchFilter, error) {
```

Comment:
same here, is it necessary to define `NewSearchFilter` an exported function

Reply:
done